platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -113,7 +113,7 @@ public InternalProblem build() {
 
         Throwable exceptionForProblemInstantiation = getExceptionForProblemInstantiation();
         if (problemStream != null) {
-            addLocationsFromProblemStream(this.locations, exceptionForProblemInstantiation);
+            addLocationsFromProblemStream(collectStackLocation ? this.locations: this.contextLocations, exceptionForProblemInstantiation);
```

Comment:
```suggestion
            addLocationsFromProblemStream(collectStackLocation ? this.locations: this.contextLocations, exceptionForStackLocation);
```

💅  The name can now be better.

Reply:
Fixed